specs/FrameProcesssInfo.md

```diff
@@ -0,0 +1,503 @@
+/// in the `GetProcessInfosWithDetails` method call.
+[uuid(8e7d154c-e2ca-11ed-b5ea-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2GetProcessInfosWithDetailsCompletedHandler : IUnknown {
+  HRESULT Invoke([in] HRESULT errorCode, [in] ICoreWebView2ProcessInfoCollection* value);
+}
+
+/// A continuation of the ICoreWebView2ProcessInfo interface.
+[uuid(982ae768-e2ca-11ed-b5ea-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2ProcessInfo2 : ICoreWebView2ProcessInfo {
+  /// The collection of associated `FrameInfo`s which are actively running
+  /// (showing UI elements) in this renderer process. `AssociatedFrameInfos`
+  /// will only be populated when obtained via calling 
+  /// `CoreWebView2Environment.GetProcessInfosWithDetails` and when this 
+  /// `CoreWebView2ProcessInfo` corresponds to a renderer process.
+  /// `CoreWebView2ProcessInfo` objects obtained via `CoreWebView2Environment.
+  /// GetProcessInfos` or for non-renderer processes will always have an empty
+  /// `AssociatedFrameInfos`. The `AssociatedFrameInfos` may also be empty for
+  /// renderer processes that have no active frames.
+  ///
+  /// \snippet ProcessComponent.cpp AssociatedFrameInfos
+  [propget] HRESULT AssociatedFrameInfos(
+    [out, retval] ICoreWebView2FrameInfoCollection** frames);
+}
+
+/// A continuation of the ICoreWebView2Environment13 interface.
+[uuid(9d4d8624-e2ca-11ed-b5ea-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2Environment14 : ICoreWebView2Environment13 {
+  /// Gets a snapshot collection of `ProcessInfo`s corresponding to all currently
+  /// running processes associated with this `CoreWebView2Environment` except 
+  /// for crashpad process. This provides the same list of `ProcessInfo`s as 
+  /// what's provided in `GetProcessInfos`, but additionally provides a list of 
+  /// associated `FrameInfo`s which are actively running (showing UI elements) 
+  /// in the renderer process. See `AssociatedFrameInfos` for more information.
+  /// 
+  /// \snippet ProcessComponent.cpp GetProcessInfosWithDetails
+  HRESULT GetProcessInfosWithDetails([in] ICoreWebView2GetProcessInfosWithDetailsCompletedHandler* handler);
+}
+
+/// A continuation of the ICoreWebView2FrameInfo interface.
+[uuid(a7a7e150-e2ca-11ed-b5ea-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2FrameInfo2 : ICoreWebView2FrameInfo {
+  /// This frame's parent frame's `FrameInfo`. `ParentFrameInfo` will only be
+  /// populated when obtained via calling 
+  /// `CoreWebView2ProcessInfo.AssociatedFrameInfos`. 
+  /// `CoreWebView2FrameInfo` objects obtained via `CoreWebView2.ProcessFailed` will
+  /// always have a `null` `ParentFrameInfo`. This property is also `null` for the  
+  /// top most document in the WebView2 which has no parent frame.
+  /// Note that this `ParentFrameInfo` could be out of date as it's a snapshot.   

```

Comment:
How are developers expected to check if this frame info is out of date? Track whether the ICoreWebView2Frame with same ID was destroyed?

Reply:
Developers can check if the frame info is up-to-date by ID, but it's only limited on CoreWebView2 and first sub layer Iframe CoreWebViewFrame. Developers will make their own strategies to call this API. For example, I've heard a customer will check it periodically.

Reply:
Please document: does the snapshot correspond to when the call is made or when the call completes? And note you can use frame created/destroyed to know if the snapshot will correspond to the live data on CoreWebView2(Frame)